src/aggregate/aggregate_request.c

```diff
@@ -568,7 +568,13 @@ static int parseQueryArgs(ArgsCursor *ac, AREQ *req, RSSearchOptions *searchOpts
   searchOpts->ninkeys = inKeys.argc;
   searchOpts->legacy.infields = (const char **)inFields.objs;
   searchOpts->legacy.ninfields = inFields.argc;
-  searchOpts->language = RSLanguage_Find(languageStr, 0);
+  // if language is NULL, set it to RS_LANG_INDEX_DEFAULT and it will be updated
+  // later, taking the index language
+  if(languageStr == NULL) {
+    searchOpts->language = RS_LANG_INDEX_DEFAULT;
+  } else {
+    searchOpts->language = RSLanguage_Find(languageStr, 0);
```

Comment:
Can `RSLanguage_Find` return `RS_LANG_UNSET` on NULL?

What about LLAPI (low-level API) such as `RediSearch_CreateDocument`,
Need to also add the same logic there?

Reply:
No, `RS_LANG_UNSET` is used only to identify that the user did not pass any language in `FT.QUERY` command but the logic of `RSLanguage_Find` was not modified, and should not return `RS_LANG_UNSET.
As `RSLanguage_Find` was not modified, I don't see any problem with the LLAPI or need to change it.

Reply:
LLAPI is working, the change was done in `redisearch_api.c`, if the index contains a language different from default, it is assigned to `RSSearchOptions->language`.
Two tests were created:
* `TEST_F(LLApiTest, testIndexWithDefaultLanguage)`
* `TEST_F(LLApiTest, testIndexWithCustomLanguage)`